File: swing/src/main/java/wannabe/swing/renderer/FilledRoundedSquare.java
package wannabe.swing.renderer;

import java.awt.Graphics;
import wannabe.Projected;

public class FilledRoundedSquare extends SwingRenderer {
  @Override public void draw(Graphics g, Projected r) {
    int arc = r.size / 3;
    g.fillRoundRect(r.left, r.top, r.size, r.size, arc, arc);
  }
}

File: swing/src/main/java/wannabe/swing/renderer/FilledRoundedSquare.kt
package wannabe.swing.renderer

import wannabe.swing.SwingProjected
import java.awt.Graphics

class FilledRoundedSquare : SwingRenderer() {
  override fun draw(
    g: Graphics,
    p: SwingProjected
  ) {
    val arc = p.size / 3
    g.fillRoundRect(p.left, p.top, p.size, p.size, arc, arc)
  }
}
